src/wombat.js

```diff
@@ -2904,25 +2910,25 @@ Wombat.prototype.overrideHistoryFunc = function(funcName) {
 
     orig_func.call(this, stateObj, title, rewritten_url);
 
-    var origTitle = wombat.$wbwindow.document.title;
+    var origTitle = historyWin.document.title;
 
     if (wombat.WBAutoFetchWorker) {
-      wombat.$wbwindow.setTimeout(function() {
-        if (!title && wombat.$wbwindow.document.title !== origTitle) {
-          title = wombat.$wbwindow.document.title;
+      historyWin.setTimeout(function() {
+        if (!title && historyWin.document.title !== origTitle) {
+          title = historyWin.document.title;
         }
 
         wombat.WBAutoFetchWorker.fetchAsPage(rewritten_url, resolvedURL, title);
       }, 100);
     }
 
-    wombat.sendHistoryUpdate(resolvedURL, title);
+    wombat.sendHistoryUpdate(resolvedURL, title, historyWin);
   };
 
   this.$wbwindow.history[funcName] = rewrittenFunc;
-  if (this.$wbwindow.History && this.$wbwindow.History.prototype) {
-    this.$wbwindow.History.prototype[funcName] = rewrittenFunc;
-  }
+  //if (this.$wbwindow.History && this.$wbwindow.History.prototype) {
```

Comment:
I think it would be best to __not__ comment is part out since leaves the un-overridden version available via History's prototype. 

Each objects prototype is unique to the execution context it exists in (https://heycam.github.io/webidl/#es-environment).

Also maybe a good idea to set `___wb_ownWindow` on the prototype of History.

```suggestion
  if (this.$wbwindow.History && this.$wbwindow.History.prototype) {
    if (!this.$wbwindow.History.prototype.___wb_ownWindow) {
      this.$wbwindow.History.prototype.___wb_ownWindow = this.$wbwindow;
    }
    this.$wbwindow.History.prototype[funcName] = rewrittenFunc;
  } else if (!this.$wbwindow.history.___wb_ownWindow) {
    this.$wbwindow.history.___wb_ownWindow = this.$wbwindow;
  }
```